src/classes/RD2_OpportunityEvaluation_BATCH.cls

```diff
@@ -0,0 +1,137 @@
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Recurring Donations
+* @description Recurring Donations Bulk Processing Batch Job to review EVERY Recurring Donation
+* record to evaluate its attached Opportunities to take the necessary action.
+*
+*/
+public without sharing class RD2_OpportunityEvaluation_BATCH
+    implements Database.Batchable<SObject>, Database.Stateful, Schedulable {
+
+    /**
+    * @description Batch size for Apex Job. Uses default if the Custom Setting batch size is not provided.
+    */
+    @TestVisible
+    public Integer batchSize {
+        get {
+            if (batchSize == null) {
+                Decimal configValue = UTIL_CustomSettingsFacade.getRecurringDonationsSettings().Recurring_Donation_Batch_Size__c;
+
+                if (configValue == null) {
+                    configValue = 50;
+                }
+
+                batchSize = configValue.intValue();
+            }
+            return batchSize;
+        }
+        private set;
+    }
+
+    /*******************************************************************************************************
+    * @description Count of RD's processed by the batch
+    */
+    private Integer numberRDsProcessed = 0;
+
+    /*******************************************************************************************************
+    * @description Count of RD's that hit errors while processing
+    */
+    private Integer numberOfFailures = 0;
+
+    /**
+     * @description Schedulable Execute Method to launch the Batch Job
+     */
+    public void execute(SchedulableContext context) {
+        Database.executeBatch(new RD2_OpportunityEvaluation_BATCH(), batchSize);
+    }
+
+
+    /**
+    * @description Batch Start Method
+    */
+    public Database.QueryLocator start(Database.BatchableContext context) {
+        if (RD2_EnablementService.isRecurringDonations2Enabled) {
+            return Database.getQueryLocator(getRDBatchQuery());
+        } else {
+            throw new RecurringDonationsBatchException(System.Label.RD_ErrorRD2BatchJobCannotBeRun);
+        }
+    }
+
+    /**
+    * @description Batch Execute Method
+    */
+    public void execute(Database.BatchableContext context, List<npe03__Recurring_Donation__c> rds) {
+
+        Set<Id> rdIds = (new Map<Id, npe03__Recurring_Donation__c>(rds)).keySet();
+
+        RD2_OpportunityEvaluationService oppEvalService = new RD2_OpportunityEvaluationService()
+            .withRds(rdIds);
+        oppEvalService.evaluateOpportunities();
+
+        numberRDsProcessed += rds.size();
+    }
+
+    /**
+    * @description Build the query of all standard RD fields to use for the batch query
+    * @return SOQL
+    */
+    @TestVisible
+    private String getRDBatchQuery() {
+        Set<String> rdFields = new Set<String>{
+            'Id'
+        };
+
+        return new UTIL_Query()
```

Comment:
All RD's are selected to ensure that an RD that might have been changed to Closed, but still has Open opps, is properly processed. That could happen in a bulk data load or bulk update scenario where triggers on the RD were disabled or unable to run due to bulk constraints (i.e., gov limits).

Sorting is not guaranteed in a batch query locator, so querying all Active Schedules, sorted by RecurringDonationId will not give us what we actually need to process the Opps. Not to mention, it will return more records than there are RD's, and the individual batches might cause schedule records for a single RD to be split across multiple batches.